esp/services/esdl_svc_engine/esdl_monitor.cpp

```diff
@@ -518,6 +518,31 @@ class CEsdlMonitor : implements IEsdlMonitor, public CInterface, implements IEsd
                     bindingtree->setProp("@port", port);
                     servicetree->setProp("@name", serviceName.str());
                     servicetree->setProp("@type", "DynamicESDL");
+
+                    // Copy resource_map properties from the desdlBindingConfig section for this
+                    // dynamic service into the bindingtree in the expected format
+                    IPropertyTree* desdlBindingConfig = bindingtree->queryPropTree("desdlBindingConfig");
+                    if (desdlBindingConfig)
+                    {
+                        IPropertyTree* bindingAuthenticate = bindingtree->queryPropTree("Authenticate");
+                        xpath.setf("resource_map/%s", serviceName.str());
+                        IPropertyTree* resourceMap = desdlBindingConfig->queryPropTree(xpath.str());
+                        if (resourceMap && bindingAuthenticate)
+                        {
+                            mergePTree(bindingAuthenticate, resourceMap);
+                            desdlBindingConfig->removeProp(xpath.str());
+                        }
+                    }
+
+                    // Likewise copy over any additional dynamic service-specific binding configuration
+                    xpath.setf("additionalInfo/%s", serviceName.str());
+                    IPropertyTree* addtlInfo = desdlBindingConfig->queryPropTree(xpath.str());
```

Comment:
I think this block should be inside the 
```
if (desdlBindingConfig)
{
```
otherwise desdlBindingConfig might be null here when you access queryPropTree()?